proposals/0013-expanded-comparison-sampling.md

```diff
@@ -0,0 +1,432 @@
+ that currently support the non-comparison methods:
+
+* `SampleCmpBias(SamplerComparisonState S, float<num> Location, float Bias)`
+  performs a comparison sample using the implicitly-determined LOD with the
+  explicitly-provided _Bias_ value added to that result.
+* `SampleCmpGrad(SamplerState S, float<num> Location, float DDX, float DDY)`
+   samples using the LOD calculated using the explicitly-provided
+   _DDX_ and _DDY_ gradients.
+
+Only the simplest overloads of the new and existing builtin methods are
+ described in detail here for the sake of brevity,
+Though not included above, all relevant overloads of `SampleBias` and
+ `SampleGrad` will be reflected in their new `SampleCmpBias` and
+ `SampleCmpGrad` counterparts.
+The parameters not explicitly described here will function exactly as they do
+ when used by the non-comparison builtin methods:
+
+* `SampleCmpBias(SamplerComparisonState S, float<num> Location, float Bias)`
+* `SampleCmpBias(SamplerComparisonState S, float<num> Location, float Bias, int<num> Offset)`
+* `SampleCmpBias(SamplerComparisonState S, float<num> Location, float Bias, int<num> Offset, float Clamp)`
+* `SampleCmpBias(SamplerComparisonState S, float<num> Location, float Bias, int<num> Offset, float Clamp, uint Status)`
+* `SampleCmpGrad(SamplerState S, float<num> Location, float DDX, float DDY)`
+* `SampleCmpGrad(SamplerState S, float<num> Location, float DDX, float DDY, int<num> Offset)`
+* `SampleCmpGrad(SamplerState S, float<num> Location, float DDX, float DDY, int<num> Offset, float Clamp)`
+* `SampleCmpGrad(SamplerState S, float<num> Location, float DDX, float DDY, int<num> Offset, float Clamp, uint Status)`
+
+To add LOD calculation for comparison samplers, HLSL will introduce and require
+ these new comparison sampler method overloads to all texture objects that
+ currently support the non-comparison method overloads:
+
+* `CalculateLevelOfDetailUnclamped(SamplerComparisonState S, float<num> Location)`
+  returns the raw LOD value determined implicitly using gradients representing
+  how much the _Location_ changes across neighboring pixels regardless of
+  whether it falls outside the valid range.
+* `CalculateLevelOfDetail(SamplerComparisonState S, float<num> Location)`
+  returns the implicitly-determined LOD value clamped to the valid range.
+
+## Detailed Design
+
+### HLSL additions
+
+These new HLSL texture methods will be added:
+
+```c++
+<Format> <TexObject>::SampleCmpGrad(
+    SamplerComparisonState S,
+    float<num> Location,
+    float<num> DDX,
+    float<num> DDY,
+    float CompareValue,
+    [int<num> Offset,]
+    [float Clamp,]
+    [out uint Status]
+);
+<Format> TextureCube::SampleCmpGrad(
+    SamplerComparisonState S,
+    float<num> Location,
+    float<num> DDX,
+    float<num> DDY,
+    float CompareValue,
+    [float Clamp,]
+    [out uint Status]
+);
+```
+
+Where `TexObject` represents any texture object type except `Texture2DMS*`,
+ `Texture3D`, and `TextureCube*` objects.
```

Comment:
Perhaps we should list the allowed texture types, instead of the exclusions?

Reply:
Yea, I would prefer that we explicitly state "we're adding method X to type Y" for each type we are changing rather than tying to have shorter but less explicit language.

In the future if we add a new `Texture4D` type, this language would be confusing to read, but an explicit list of the types that are changed here will stay more accurate.

Reply:
done